### PR TITLE
#185218929 Users should be able to chat

### PR DESCRIPTION
**Pull request by nishimweprince**
## What does this PR do?

Adds chat and chat room features to our ECommerce application.

## Description of Task to be completed?

+ Given a user (buyer) is logged in and online, they should be able to share a public
+ Given a user opens the chat tab, all fellow active users are notified of a new party joining the chat
+ When a user sends a message, it is echoed on all active participants
+ All messages are stored in the database
+ A user should be able to read all past messages from the chat

## How should this be manually tested?

- Clone the repository using `git clone https://github.com/atlp-rwanda/e-comm-team-techsmith-fn`

- Checkout to the branch using `git checkout ft-chatroom-185218929`

- Install node packages that the application depends on: `npm i`

- Create a `.env` file in the root directory and add all environment variables required to run the application. The required variables are listed in the `.env.example` file available in root directory of the application.

- Start the application development server: `npm start`

- Check out the components documentation: `npm run storybook`

## What are the relevant pivotal tracker/Trello stories?

+ [#185218929](https://www.pivotaltracker.com/story/show/185218929)

## Screenshots (if applicable)
Screenshot 2023-06-16 at 15 37 26
Screenshot 2023-06-17 at 17 25 09
Screenshot 2023-06-17 at 17 25 47
Screenshot 2023-06-17 at 17 25 58
Screenshot 2023-06-17 at 17 27 34

### CODE REVIEW

File: src/containers/SignupContainer.jsx
Comment (bergad37) [inline src/containers/SignupContainer.jsx:1]: Commented line 97 and 120 could be removed!

File: src/styles/index.scss
Comment (bergad37) [inline src/styles/index.scss:1]: follow the naming convention  for naming scss files chatroom instead of Chatroom!

Comment (nishimweprince) [reply src/styles/index.scss:1]: Sure, let me do that. Thanks!

File: src/states/features/auth/authServices.js
Comment (irakevine) [inline src/states/features/auth/authServices.js:1]: remove line 16., that you know there is no console we are allowed to have in each file

File: src/components/Chatroom/ChatBox.jsx
Comment (irakevine) [inline src/components/Chatroom/ChatBox.jsx:1]: If you don't mind please explain to me the purpose of this file and what will help us,
thank you

**Review by mken1205 — COMMENTED** [review]
Some of these files don't have a new line at the end

File: src/states/features/auth/authServices.js
Comment (bergad37) [reply src/states/features/auth/authServices.js:1]: Remove the console here!

File: vite.config.js.timestamp-1686515408485-7ded070124dfd.mjs
Comment (cyubahiro367) [inline vite.config.js.timestamp-1686515408485-7ded070124dfd.mjs:1]: why do u delete vite config file?

File: src/states/features/auth/authServices.js
Comment (cyubahiro367) [inline src/states/features/auth/authServices.js:1]: console logs are not allowed

Comment (nishimweprince) [reply src/states/features/auth/authServices.js:1]: Sure

File: vite.config.js.timestamp-1686515408485-7ded070124dfd.mjs
Comment (nishimweprince) [reply vite.config.js.timestamp-1686515408485-7ded070124dfd.mjs:1]: I didn't find it necessary to have it in our codebase as we have Vite configuration in a separate file.

File: src/states/features/auth/authServices.js
Comment (nishimweprince) [reply src/states/features/auth/authServices.js:1]: This was for the sake of 2FA, but let me remove it.

File: src/components/Chatroom/ChatBox.jsx
Comment (nishimweprince) [reply src/components/Chatroom/ChatBox.jsx:1]: This is the main container that holds the (three) sections for available rooms, user messages, and all active participants. Refer to this image for better context.
Image showing chatbox

**Review by jcharles921 — APPROVED** [review]

**Comment by nishimweprince** [comment]
> Some of these files don't have a new line at the end

I have fixed that too!

**Review by jcharles921 — APPROVED** [review]
LGTM !

File: src/components/Loading.jsx
Comment (irakevine) [inline src/components/Loading.jsx:1]: I like the way you did this loading instead of using any other one and using the same one we used

**Review by Gabriel-mdv — APPROVED** [review]
It works well.. good job. I used your .env to test; however, add those needed environment variables in .env.example , so that others can know what to look and ask for.

**Review by irakozejules — APPROVED** [review]

**Review by frankflej — COMMENTED** [review]
Your work looks clean you need to rebase so that your PR gets to be merged

**Review by mken1205 — APPROVED** [review]

**Review by frankflej — APPROVED** [review]
Good work

**Review by ReneNtabana — APPROVED** [review]

**Review by irakevine — APPROVED** [review]
Great Work

File: src/components/Loading.jsx
Comment (nishimweprince) [reply src/components/Loading.jsx:1]: Thanks @irakevine 🙏🏽

**Review by niyontwali — APPROVED** [review]

**Pull request merged**